test/test.js

```diff
@@ -1319,6 +1319,20 @@ describe('node-fetch', () => {
 		});
 	});
 
+	it('should create http.Agent class if options.agent is provided as an object', function() {
```

Comment:
upon further study, I'm sticking with my initial comment, since `https.agent` inherits `http.agent` and the added logic is only concerned with initializing the objects.

I do think adding an `https.agent` test in general is great, but that is outside the scope of this PR and can be made into a separate branch & PR